pkg/kube/configmapwatcher/configmapwatcher.go

```diff
@@ -44,8 +44,7 @@ type Controller struct {
 	configMapName      string
 	callback           func(*v1.ConfigMap)
 
-	mu        sync.RWMutex
-	hasSynced bool
+	hasSynced atomic.Bool
```

Comment:
I donot think this is right, the atomic.Bool can not be copied. So this must be a pointer type